sdk_container/src/third_party/coreos-overlay/coreos-base/oem-ec2/files/manglefs.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -euo pipefail
+
+rootfs="${1}"
+
+to_delete=(
+  /etc/systemd/system/amazon-ssm-agent.service
+  /etc/systemd/system/multi-user.target.requires/coreos-metadata-sshkeys@core.service
+  /etc/amazon/ssm/
+  /etc/eks/bootstrap.sh
+)
+
+rm -rf "${to_delete[@]/#/${rootfs}}"
+
+mkdir -p "${rootfs}/usr/lib/systemd/system/amazon-ssm-agent.service.d"
+cat > "${rootfs}/usr/lib/systemd/system/amazon-ssm-agent.service.d/10-bindpaths.conf" <<-'EOF'
+[Service]
+BindPaths=/usr/share/amazon/ssm/:/etc/amazon/ssm/ /usr/share/amazon/eks/boostrap.sh:/etc/eks/bootstrap.sh
+EOF
+
+mkdir -p "${rootfs}/usr/lib/systemd/system/multi-user.target.d"
+{ echo "[Unit]"; echo "Upholds=amazon-ssm-agent.service coreos-metadata-sshkeys@core.service"; } > "${rootfs}/usr/lib/systemd/system/multi-user.target.d/10-oem-ec2.conf"
```

Comment:
Can we do the starting of `coreos-metadata-sshkeys@core.service` in the base image - we also do the coreos-cloudinit start in the base image and could do it similarly (have a unit that has a condition for the the OEM kernel cmdline argument and then uses `Upholds=` to start it).

Reply:
Something like this
```
[Unit]
ConditionKernelCommandLine=|ignition.platform.id=packet
ConditionKernelCommandLine=|flatcar.oem.id=packet
ConditionKernelCommandLine=|coreos.oem.id=packet

ConditionKernelCommandLine=|ignition.platform.id=ec2
ConditionKernelCommandLine=|flatcar.oem.id=ec2
ConditionKernelCommandLine=|coreos.oem.id=ec2

ConditionKernelCommandLine=|ignition.platform.id=digitalocean
ConditionKernelCommandLine=|flatcar.oem.id=digitalocean
ConditionKernelCommandLine=|coreos.oem.id=digitalocean

ConditionKernelCommandLine=|ignition.platform.id=gce
ConditionKernelCommandLine=|flatcar.oem.id=gce
ConditionKernelCommandLine=|coreos.oem.id=gce

Upholds=coreos-metadata-sshkeys@core.service

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/bin/true

[Install]
WantedBy=multi-user.target
```

Reply:
```suggestion
{ echo "[Unit]"; echo "Upholds=amazon-ssm-agent.service"; } > "${rootfs}/usr/lib/systemd/system/multi-user.target.d/10-oem-ec2.conf"
```

Reply:
What are the contents of `/usr/lib/systemd/system/multi-user.target.d/10-oem-ec2.conf`? Is this really the most recent state?

Reply:
Ah, understood, so it seems the condition is only for the [Service] section? And Upholds= still gets used all the time. Then yes, `ExecStart=systemctl start coreos-metadata-sshkeys@core.service` instead of `ExecStart=/bin/true` sounds good!

Reply:
Ok, so more follow-up for the init PR… Sorry for the misleading suggestion